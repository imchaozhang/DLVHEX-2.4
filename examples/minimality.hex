nat(0).
nat(1).
nat(2).

s :- &testEven[m,n].

% spoil m/n
m(X) :- s, nat(X).
n(X) :- s, nat(X).

% also spoil r/p
r(X) :- s, nat(X).
p(X) :- s, nat(X).

r(0) v r(2).
p(0) v p(1).

% create as many m/n instances as given by p/r
m(X) :- nat(X), X<=Y, p(Y).
n(X) :- nat(X), X<=Y, r(Y).
