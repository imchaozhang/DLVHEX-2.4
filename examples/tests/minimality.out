{p(1), r(0), n(0), m(0), m(1), nat(0), nat(1), nat(2)}
{p(1), r(2), n(0), n(1), n(2), m(0), m(1), nat(0), nat(1), nat(2)}
